homeassistant/components/switchbot/cover.py

```diff
@@ -0,0 +1,141 @@
+"""Support for SwitchBot curtains."""
+from typing import Any, Dict
+
+# pylint: disable=import-error, no-member
+import switchbot
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_CURTAIN,
+    PLATFORM_SCHEMA,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    CoverEntity,
+)
+from homeassistant.const import CONF_MAC, CONF_NAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.restore_state import RestoreEntity
+
+COVER_FEATURES = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
+
+CLOSED_POSITION = 0
+OPEN_POSITION = 100
+
+DEFAULT_NAME = "Switchbot Curtain"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler